app/queries/consumer_apps/rpush_app_query.rb

```diff
@@ -18,24 +18,56 @@ def initialize(app_bundle:, platform:)
     def call
       if consumer_app.android?
         android_app = Rpush::Gcm::App.where(name: app_name).first
-        android_app || recreate_android_app!
+        return android_app unless stale?(rpush_app: android_app)
+
+        recreate_android_app!
       elsif consumer_app.ios?
         ios_app = Rpush::Apns2::App.where(name: app_name).first
-        ios_app || recreate_ios_app!
+        return ios_app unless stale?(rpush_app: ios_app)
+
+        recreate_ios_app!
       end
     end
 
     private
 
     attr_reader :app_bundle, :app_name, :consumer_app, :platform
 
+    # Returns whether the app is stale, which means it needs to be
+    # created or recreated if it already exists (credentials changed)
+    def stale?(rpush_app:)
+      # If the app doesn't exist => stale
+      return true if rpush_app.nil?
+
+      # If credentials have changed on either platform => stale (destroy + recreate)
+      if consumer_app.android? && rpush_app.auth_key != app_auth_credentials
+        rpush_app.destroy
+        return true
+      elsif consumer_app.ios? && rpush_app.certificate != app_auth_credentials
+        rpush_app.destroy
+        return true
+      end
+
+      # App exists and credentials still match => not stale
+      false
+    end
```

Comment:
This method is what we'll rely on to know when to create an rpush app, or destroy the stale one + recreate it using the updated credentials regardless of where they come from.